utils/zerovec/src/zerovec/mod.rs

```diff
@@ -80,55 +83,46 @@ use core::ops::Deref;
 /// Benchmarks can be found in the project repository, with some results found in the [crate-level documentation](crate).
 ///
 /// See [the design doc](https://github.com/unicode-org/icu4x/blob/main/utils/zerovec/design_doc.md) for more details.
-#[derive(Clone)]
-#[non_exhaustive]
-pub enum ZeroVec<'a, T>
+pub struct ZeroVec<'a, T>
 where
-    T: AsULE + ?Sized,
+    T: AsULE,
```

Comment:
Question: Why did you drop the `?Sized` ?

Reply:
It's completely useless, this is for Copy stack types, there's no way to have a `?Sized` type in ULE and have it work. The ULE methods won't be callable for such types.

Reply:
But this is the AsULE. I can't think of a specific use case, but it doesn't seem that T: Sized needs to be a requirement

Reply:
AsULE returns owned values, so to do anything useful it has to be Sized

In general the idea is to be as generic as possible in trait bounds where more specific bounds are unnecessary, but I don't think that advice applies for opt out bounds like `?Sized`, "be as generic as possible" is shorthand for "don't use unnecessary bounds"